File: Python/reverse.py
input_string = input("Enter array elements (space-separated): ")
arr = [int(x) for x in input_string.strip().split()]
n = len(arr)
k = 4  

start = 0
end = n - 1
while start < end:
    temp = arr[start]
    arr[start] = arr[end]
    arr[end] = temp
    start += 1
    end -= 1

print("\nAfter full array reverse:", arr)

File: Mathematical_Basics/reverse.java
import java.util.Scanner;
public class reverse {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of elements in the array: ");
        int n = sc.nextInt();
        int arr[]= new int[n],end = arr.length - 1;
        int start = 0;
        System.out.println("Enter the elements of the array (space-separated):");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }

        while (start < end) {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
        System.out.print("\nReversed array: ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    sc.close();
        System.out.println(); // For better output formatting
    }
}
